---
title: "SeedPAD results"
author: "David Castro"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_float: true
    number_sections: true
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir=getwd())
```

# Libraries required for the analisys
```{r, echo =TRUE, message= FALSE}
library(picante)
library(phyloseq) #qiime outputs
library(vegan) #multivariate statistics 
library(DESeq2) #differential selected fungi
library(tidyverse)
library(pheatmap)
library(plyr)
library(ComplexHeatmap)
library(circlize)
library(FSA)
library(rcompanion)
library(genefilter)
library(ggplot2)
library(bibtex)
library(VennDiagram)
library(pairwiseAdonis)
library(rwantshue) #color generator
library(gplots)
library(survival) #survival statistic analysis
library(survminer) #survival plots
```

# Site metadata
## Climate
Climate data was imported from SMHI database from Frösön station, located approximatelly 150 km west-southwest from the clear-cut. Normal precipitation (blue bars), calculated with data from 1961 to 1990 compared with the precipitation of 2017 (orange bars) suggest a variable year, with no tendency of a dry or wet year. Normal temperature (green line), have no much difference with the mean temperature (dashed red line).

```{r echo= FALSE}
temp_pp <- read.csv("growth/temp_pp.csv", header = TRUE, sep = ';', dec = ',')
head(temp_pp)
temp_pp$Month <- factor(temp_pp$Month, levels=c('June','July','August','September','October'))

temp_pp

#pdf("Manuscript/Plots/PP_Temp.pdf",height=10,width=10)
ggplot(temp_pp, aes(x = Month))+
  geom_bar(aes(y = pp),stat = "identity", fill = 'orange', width = 0.4, position = position_nudge(x=.22))+
  geom_bar(aes(y = pp_normal),stat = "identity", fill = 'blue', width = 0.4, position = position_nudge(x=-.22))+
  geom_line(aes(y=T_Mean*8, group = 1), colour = 'darkred', size= 1, linetype = "dashed")+
  geom_line(aes(y=T_Normal*8, group = 1), colour = 'darkgreen', size= 1)+
  geom_line(aes(y=Min_mean*8, group = 1), colour = 'grey', size= 1)+
  geom_line(aes(y=Max_mean*8, group = 1), colour = 'red', size= 1)+
  scale_y_continuous(sec.axis = sec_axis(~./8, name = "Temperature (C°)"))+
  labs(y = "Precipitation (mm)")+
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        axis.line.x.bottom = element_line(size = 1, colour = "black"),
        axis.line.y = element_line(size = 1, colour = "black"),
        panel.background = element_blank(),
        panel.grid = element_line(size = 0.5, colour = "gray"))
```

## Carbon & Nitrogen
```{r}
cn <- read.csv("growth/CN_content.csv", header = TRUE, sep = ';',dec = ",")
head(cn)

cn$Treatment <- factor(cn$Treatment, levels=c('Control','Arginine_P','Osmocote'))
cn$Sample_type <- factor(cn$Sample_type, levels=c('Needles','Ectorhizosphere','Clear_cut'))

cn$N <- cn$N*10 #transform from gN/gSoil into mgN/gSoil
cn$C <- cn$C*10 #transform from gC/gSoil into mgC/gSoil
```

### CC vs Ectorhizosphere
#### Sample effect
```{r}
cn_soil_ecto=cn %>% filter(Sample_type != "Needles")

#pdf("Manuscript/Plots/N_content.pdf")
ggplot(cn_soil_ecto, aes(x = Sample_type, y = N, fill = Sample_type)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("goldenrod4", "firebrick4")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "N%", limits = c(0,4))

#pdf("Manuscript/Plots/C_content.pdf")
ggplot(cn_soil_ecto, aes(x = Sample_type, y = C, fill = Sample_type)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("goldenrod4", "firebrick4")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "C%", limits = c(0,130))

#pdf("Manuscript/Plots/CN_content.pdf")
ggplot(cn_soil_ecto, aes(x = Sample_type, y = CN_ratio, fill = Sample_type)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("goldenrod4", "firebrick4")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "CN ratio")
```

```{r}
aggregate(cn_soil_ecto$C, list(cn_soil_ecto$Sample_type), median)
kruskal.test(C ~ Sample_type, data = cn_soil_ecto) #non significant

aggregate(cn_soil_ecto$N, list(cn_soil_ecto$Sample_type), median)
kruskal.test(N ~ Sample_type, data = cn_soil_ecto) #non significant

aggregate(cn_soil_ecto$CN_ratio, list(cn_soil_ecto$Sample_type), median)
kruskal.test(CN_ratio ~ Sample_type, data = cn_soil_ecto) #non significant
```

#### Treatment effect
```{r}
#pdf("Manuscript/Plots/N_content.pdf")
ggplot(cn_soil_ecto, aes(x = Sample_type, y = N, fill = Treatment)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("green", "red","blue")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "N%", limits = c(0,4))


#pdf("Manuscript/Plots/C_content.pdf")
ggplot(cn_soil_ecto, aes(x = Sample_type, y = C, fill = Treatment)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("green", "red","blue")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "C%", limits = c(0,130))


#pdf("Manuscript/Plots/CN_content.pdf")
ggplot(cn_soil_ecto, aes(x = Sample_type, y = CN_ratio, fill = Treatment)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("green", "red","blue")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "CN ratio")
```

```{r}
aggregate(cn_soil_ecto$C, list(cn_soil_ecto$Treatment), median)
kruskal.test(C ~ Treatment, data = cn_soil_ecto) #non significant

aggregate(cn_soil_ecto$N, list(cn_soil_ecto$Treatment), median)
kruskal.test(N ~ Treatment, data = cn_soil_ecto) #non significant

aggregate(cn_soil_ecto$CN_ratio, list(cn_soil_ecto$Treatment), median)
kruskal.test(CN_ratio ~ Treatment, data = cn_soil_ecto) #non significant
```

# Survival rate

The survival results were obtained by counting all seedlings in October 2017. Number 0 represents all seedlings that were dead and number 1 all living seedlings

Reading the .csv

```{r, echo= FALSE}
survival <- read.csv("growth/survival_rate.csv", header = TRUE, sep = ';')
head(survival)
```
```{r}
survival$Treatment <- factor(survival$Treatment, levels=c('Control','Arginine','Osmocote'))
survival$Status_2 <- ifelse(survival$Status==0,1,0) #'Censoring' status
survival$Time <- 1

# Kaplan-Meier curve
surv_object <- Surv(time=survival$Time, survival$Status_2)
#fit1 <- survfit(surv_object ~ Treatment, data= survival)

#Cox proportional hazards
fit.coxph <- coxph(surv_object ~ Treatment, data = survival)
fit.coxph

survival <- data.frame(Treatment=c("Control","Arginine_P","Osmocote") ,  Percentage=c(35,53,50))

survival$Treatment <- factor(survival$Treatment, levels=c('Control','Arginine_P','Osmocote'))

ggplot(survival,aes(Treatment, Percentage, fill = Treatment)) + 
  geom_bar(stat = "identity", colour="black") + scale_fill_manual(values = c("green", "red", "blue")) +
  theme_bw() +
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Survival rate", limits = c(0,100))
```

## C and N content needles
### Treatment effect
```{r}
cn_needles <- cn %>% filter(Sample == "Needles")

#pdf("Manuscript/Plots/N_content.pdf")
ggplot(cn_needles, aes(x = Treatment, y = N, fill = Treatment)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("green", "red","blue")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "N%", limits = c(0,28))

#pdf("Manuscript/Plots/C_content.pdf")
ggplot(cn_needles, aes(x = Treatment, y = C, fill = Treatment)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("green", "red","blue")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "C%", limits = c(440,540))

#pdf("Manuscript/Plots/CN_content.pdf")
ggplot(cn_needles, aes(x = Treatment, y = CN_ratio, fill = Treatment)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("green", "red","blue")) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "CN ratio", limits = c(20,50))
```

```{r}
aggregate(cn_needles$C, list(cn_needles$Treatment), median)
kruskal.test(C ~ Treatment, data = cn_needles) 

aggregate(cn_needles$N, list(cn_needles$Treatment), median)
kruskal.test(N ~ Treatment, data = cn_needles) #non significant

aggregate(cn_needles$CN_ratio, list(cn_needles$Treatment), median)
kruskal.test(CN_ratio ~ Treatment, data = cn_needles) #non significant
```

# Growth results

This results section is made with the fresh weight of the surviving seedlings that had a root system both big enough and intact for the further DNA extraction.

Reading csv file
```{r}
growth <- read.csv("growth/seedpad_data.csv",sep = ";", dec = ",")
growth$Treatment <- factor(growth$Treatment, levels=c('Control','Arginine_P','Osmocote'))
head(growth)
```

Plotting root weight
```{r, echo=FALSE}
#pdf("Manuscript/Plots/root_weight.pdf")
ggplot(growth, aes(x = Treatment, y = Root_weight, fill = Treatment)) + 
  geom_boxplot() + scale_fill_manual(values = c("green", "red","blue"), labels = c('Control', 'Arginine phosphate', 'Ammonium nitrate')) +
  ggtitle("Root weight") +
  theme_bw() + 
  theme(plot.title=element_text(size=18, face="bold", colour="Black", hjust = .5),
        text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Weight (mg)", limits = c(0,0.20))

```

Plotting shoot weight
``` {r echo= FALSE}
#pdf("Manuscript/Plots/shoot_weight.pdf")
ggplot(growth, aes(x = Treatment, y = Shoot_weight, fill = Treatment)) + 
  geom_boxplot() + scale_fill_manual(values = c("green", "red","blue"), labels = c('Control', 'Arginine phosphate', 'Ammonium nitrate')) +
  ggtitle("Shoot weight") +
  theme_bw() + 
  theme(plot.title=element_text(size=18, face="bold", colour="Black", hjust = .5),
        text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Weight (mg)",limits = c(0,0.20))

```

Plotting root:shoot ratio. 
The root:shoot ratio was calculated by root fresh weight/shoot fresh weight.

```{r echo= FALSE}
#pdf("Manuscript/Plots/root_shoot.pdf")
ggplot(growth, aes(x = Treatment, y = Root_shoot, fill = Treatment)) + 
  geom_boxplot() + scale_fill_manual(values = c("green", "red","blue"), labels = c('Control','Arginine phosphate', 'Ammonium nitrate')) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Root:shoot ratio", limits = c(0,1.8))

```

Statistical analysis growth data. Kruskal-Wallis rank sum test with kruskal.test() function from stats package. The model used for each analysis was tissue weight~treatment type. In the case of rppt_shoot ratio it was root:shoot ratio~treatment.

```{r echo=FALSE}
aggregate(growth$Root_weight, list(growth$Treatment), median)
kruskal.test(Root_weight ~ Treatment, data = growth) #non-significant

aggregate(growth$Shoot_weight, list(growth$Treatment), median)
kruskal.test(Shoot_weight ~ Treatment, data = growth) #non-significant

aggregate(growth$Root_shoot, list(growth$Treatment), median)
kruskal.test(Root_shoot ~ Treatment, data = growth) #non-significant
```
# Amplicon sequencing analysis

The amplicon sequencing analysis was made using qiime2 for pre-processing and phyloseq package for the phylogenetic analysis.

phyloseq object preparation
```{r}
# ASV table
ASVs <- read.table("metagenomic/table_ITS.txt", sep="\t", header=TRUE, row.names=1)
# taxonomy
taxonomy <- read.table("metagenomic/taxonomy.txt", sep="\t", header=TRUE, row.names=1, na.strings=c(""," ","NA"))
taxonomy <- as.matrix(taxonomy)
# tree
tree <- read_tree("metagenomic/tree.nwk")
# sample data
metadata <- read.table("metagenomic/metadata.txt", sep="\t", header=TRUE, row.names=1, na.strings=c(""," ","NA"),dec = ",")
# phyloseq object
phy_obj <- phyloseq(otu_table(ASVs, taxa_are_rows = TRUE), 
                    tax_table(taxonomy), sample_data(metadata), tree)

tax_table(phy_obj) <- gsub(".__","",tax_table(phy_obj)) #separate taxonomy in columns

phy_obj
```

## Merging technical duplicates
First, the samples with technical replicates have to be taken from the phyloseq object, subset_samples() is used for this purpose, creating a new phyloseq object containing just the samples. Using merge_phyloseq, the replicates samples are merged into one.  
Then, the samples are merged into one with merge_samples(). Following, the samples are removed from the original phyloseq object with subset_samples(), using != (not equal to) to keep all the samples 'not equal to' the required samples. Important information that was lost during the merging process is written again in the merged phyloseq objects to later add them into the principal phy_obj()
```{r echo= FALSE}
#creating new phyloseq objects
S.arg.8.1 <- subset_samples(phy_obj,Working_ID=="S.arg.8.1")
S.arg.8.2 <- subset_samples(phy_obj,Working_ID=="S.arg.8.2")
S.arg.8 <- merge_phyloseq(S.arg.8.1,S.arg.8.2)

S.arg.9.1 <- subset_samples(phy_obj,Working_ID=="S.arg.9.1")
S.arg.9.2 <- subset_samples(phy_obj,Working_ID=="S.arg.9.2")
S.arg.9 <- merge_phyloseq(S.arg.9.1,S.arg.9.2)

S.arg.12.1 <- subset_samples(phy_obj,Working_ID=="S.arg.12.1")
S.arg.12.2 <- subset_samples(phy_obj,Working_ID=="S.arg.12.2")
S.arg.12 <- merge_phyloseq(S.arg.12.1,S.arg.12.2)

S.arg.20.1 <- subset_samples(phy_obj,Working_ID=="S.arg.20.1")
S.arg.20.2 <- subset_samples(phy_obj,Working_ID=="S.arg.20.2")
S.arg.20 <- merge_phyloseq(S.arg.20.1,S.arg.20.2)

S.arg.22.1 <- subset_samples(phy_obj,Working_ID=="S.arg.22.1")
S.arg.22.2 <- subset_samples(phy_obj,Working_ID=="S.arg.22.2")
S.arg.22 <- merge_phyloseq(S.arg.22.1,S.arg.22.2)

S.ctrl.3.1 <- subset_samples(phy_obj,Working_ID=="S.ctrl.3.1")
S.ctrl.3.2 <- subset_samples(phy_obj,Working_ID=="S.ctrl.3.2")
S.ctrl.3 <- merge_phyloseq(S.ctrl.3.1,S.ctrl.3.2)

S.ctrl.9.1 <- subset_samples(phy_obj,Working_ID=="S.ctrl.9.1")
S.ctrl.9.2 <- subset_samples(phy_obj,Working_ID=="S.ctrl.9.2")
S.ctrl.9 <- merge_phyloseq(S.ctrl.9.1,S.ctrl.9.2)

S.ctrl.10.1 <- subset_samples(phy_obj,Working_ID=="S.ctrl.10.1")
S.ctrl.10.2 <- subset_samples(phy_obj,Working_ID=="S.ctrl.10.2")
S.ctrl.10 <- merge_phyloseq(S.ctrl.10.1,S.ctrl.10.2)

S.ctrl.13.1 <- subset_samples(phy_obj,Working_ID=="S.ctrl.13.1")
S.ctrl.13.2 <- subset_samples(phy_obj,Working_ID=="S.ctrl.13.2")
S.ctrl.13 <- merge_phyloseq(S.ctrl.13.1,S.ctrl.13.2)

S.E.osm.1.1 <- subset_samples(phy_obj,Working_ID=="S.E.osm.1.1")
S.E.osm.1.2 <- subset_samples(phy_obj,Working_ID=="S.E.osm.1.2")
S.E.osm.1 <- merge_phyloseq(S.E.osm.1.1,S.E.osm.1.2)

#merging repeated samples
S.arg.8 <- merge_samples(S.arg.8, group = "Sample_type", fun = "mean")
S.arg.9 <- merge_samples(S.arg.9, group = "Sample_type", fun = "mean")
S.arg.12 <- merge_samples(S.arg.12, group = "Sample_type", fun = "mean")
S.arg.20 <- merge_samples(S.arg.20, group = "Sample_type", fun = "mean")
S.arg.22 <- merge_samples(S.arg.22, group = "Sample_type", fun = "mean")
S.ctrl.3 <- merge_samples(S.ctrl.3, group = "Sample_type", fun = "mean")
S.ctrl.9 <- merge_samples(S.ctrl.9, group = "Sample_type", fun = "mean")
S.ctrl.10 <- merge_samples(S.ctrl.10, group = "Sample_type", fun = "mean")
S.ctrl.13 <- merge_samples(S.ctrl.13, group = "Sample_type", fun = "mean")
S.E.osm.1 <- merge_samples(S.E.osm.1, group = "Sample_type", fun = "mean")

#removing repeated samples from the phyloseq object
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.8.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.8.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.9.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.9.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.12.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.12.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.20.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.20.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.22.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.arg.22.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.ctrl.3.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.ctrl.3.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.ctrl.9.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.ctrl.9.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.ctrl.10.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.ctrl.10.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.ctrl.13.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.ctrl.13.2")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.E.osm.1.1")
phy_obj <- subset_samples(phy_obj, Working_ID != "S.E.osm.1.2")

#adding useful lost information into the merged samples
sample_names(S.arg.8) <- "P12211_1061"
sample_data(S.arg.8)$Working_ID <- "S.arg.8"
sample_data(S.arg.8)$Fert_type <- "Arginine_P"
sample_data(S.arg.8)$Sample_type <- "Ectorhizosphere"
sample_data(S.arg.8)$fert_sample <- "Arginine_P.Ectorhizosphere"

sample_names(S.arg.9) <- "P12211_1063"
sample_data(S.arg.9)$Working_ID <- "S.arg.9"
sample_data(S.arg.9)$Fert_type <- "Arginine_P"
sample_data(S.arg.9)$Sample_type <- "Ectorhizosphere"
sample_data(S.arg.9)$fert_sample <- "Arginine_P.Ectorhizosphere"

sample_names(S.arg.12) <- "P12211_1067"
sample_data(S.arg.12)$Working_ID <- "S.arg.12"
sample_data(S.arg.12)$Fert_type <- "Arginine_P"
sample_data(S.arg.12)$Sample_type <- "Ectorhizosphere"
sample_data(S.arg.12)$fert_sample <- "Arginine_P.Ectorhizosphere"

sample_names(S.arg.20) <- "P12211_1076"
sample_data(S.arg.20)$Working_ID <- "S.arg.20"
sample_data(S.arg.20)$Fert_type <- "Arginine_P"
sample_data(S.arg.20)$Sample_type <- "Ectorhizosphere"
sample_data(S.arg.20)$fert_sample <- "Arginine_P.Ectorhizosphere"

sample_names(S.arg.22) <- "P12211_1078"
sample_data(S.arg.22)$Working_ID <- "S.arg.22"
sample_data(S.arg.22)$Fert_type <- "Arginine_P"
sample_data(S.arg.22)$Sample_type <- "Ectorhizosphere"
sample_data(S.arg.22)$fert_sample <- "Arginine_P.Ectorhizosphere"

sample_names(S.ctrl.3) <- "P12211_1082"
sample_data(S.ctrl.3)$Working_ID <- "S.ctrl.3"
sample_data(S.ctrl.3)$Fert_type <- "Control"
sample_data(S.ctrl.3)$Sample_type <- "Ectorhizosphere"
sample_data(S.ctrl.3)$fert_sample <- "Control.Ectorhizosphere"

sample_names(S.ctrl.9) <- "P12211_1089"
sample_data(S.ctrl.9)$Working_ID <- "S.ctrl.9"
sample_data(S.ctrl.9)$Fert_type <- "Control"
sample_data(S.ctrl.9)$Sample_type <- "Ectorhizosphere"
sample_data(S.ctrl.9)$fert_sample <- "Control.Ectorhizosphere"

sample_names(S.ctrl.10) <- "P12211_1091"
sample_data(S.ctrl.10)$Working_ID <- "S.ctrl.10"
sample_data(S.ctrl.10)$Fert_type <- "Control"
sample_data(S.ctrl.10)$Sample_type <- "Ectorhizosphere"
sample_data(S.ctrl.10)$fert_sample <- "Control.Ectorhizosphere"

sample_names(S.ctrl.13) <- "P12211_1095"
sample_data(S.ctrl.13)$Working_ID <- "S.ctrl.13"
sample_data(S.ctrl.13)$Fert_type <- "Control"
sample_data(S.ctrl.13)$Sample_type <- "Ectorhizosphere"
sample_data(S.ctrl.13)$fert_sample <- "Control.Ectorhizosphere"

sample_names(S.E.osm.1) <- "P12211_1129"
sample_data(S.E.osm.1)$Working_ID <- "S.E.osm.1"
sample_data(S.E.osm.1)$Fert_type <- "Osmocote"
sample_data(S.E.osm.1)$Sample_type <- "Soil"
sample_data(S.E.osm.1)$fert_sample <- "Osmocote.Soil"

# adding merged samples into the phyloseq object
suppressWarnings(phy_obj <- merge_phyloseq(phy_obj,S.arg.8,S.arg.9,S.arg.12,S.arg.20,S.arg.22,
                          S.ctrl.3,S.ctrl.9,S.ctrl.10,S.ctrl.13,S.E.osm.1))
```

```{r}
phy_obj # The replicated samples are now merged, reducing the number of samples
```

Filtering ASV equal or lower than 10. Filtering was made with filterfun() from the genefilter package to set the threshold and filter_taxa from phyloseq to remove the ASV lower than the set filter #revise
```{r}
flist    <- filterfun(kOverA(1, 10)) #remove AVS's equal or lower than 10 
phy_obj <- filter_taxa(phy_obj, flist, TRUE)
phy_obj
```

Filtering ASV with a minimum abundance of 0.005% per sample.
First the 0.005% had to be calculated for each sample type, by merging the phyloseq object by sample type (merge_samples()), we get the total number of reads (sample_sums()) and we can calculate the 0.005% for each. 
Then, the phyloseq object has to be splitted by sample type (subset_samples()) and the ASVs lower than the calculated 0.005% (abundance object) can be removed (prune_taxa()).
Finally, the phyloseq objects by sample type can be merged into one new phyloseq object (merge_phyloseq())
```{r}
abundance <- as.data.frame(sample_sums(merge_samples(phy_obj,group = "Sample_type")))
colnames(abundance) <- "abundance"
abundance[,".005%"] <- abundance$abundance*.00005
abundance$`.005%` <- round(abundance$`.005%`)
abundance #minimum number of reads to remove per-sample
abundance <- t(abundance)[-1,]
abundance <- as.data.frame(t(abundance))
abundance

#Subseting by sample type
Ectorhizosphere <- subset_samples(phy_obj,Sample_type=="Ectorhizosphere")
Ectorhizosphere
Root <- subset_samples(phy_obj,Sample_type=="Root")
Root
Soil <- subset_samples(phy_obj,Sample_type=="Soil")
Soil

#Removing ASV lower than 0.005% of abundance per sample type

Ectorhizosphere <- prune_taxa(taxa_sums(Ectorhizosphere) > abundance$Ectorhizosphere,Ectorhizosphere)
Ectorhizosphere
Root <- prune_taxa(taxa_sums(Root) > abundance$Root,Root)
Root
Soil <- prune_taxa(taxa_sums(Soil) > abundance$Soil,Soil)
Soil

#preparing for merge into a new phy_obj 
Ectorhizosphere <- phyloseq(otu_table(Ectorhizosphere),sample_data(Ectorhizosphere),tax_table(Ectorhizosphere))
Root <- phyloseq(otu_table(Root),sample_data(Root),tax_table(Root))
Soil <- phyloseq(otu_table(Soil),sample_data(Soil),tax_table(Soil))

phy_obj <- merge_phyloseq(Ectorhizosphere,Root,Soil)
phy_obj

#Adding phy_tree
phy_obj <- phyloseq(otu_table(phy_obj),sample_data(phy_obj),tax_table(phy_obj),tree)
phy_obj
```

Merging phylogenetic tree tips.
This is made to avoid many ASVs annotating for the same specie. Using the tip_glom() function, the tips of the phylogenetic tree lower than a given 'height' will be merged into one.
```{r}
ntaxa(phy_obj)
#pdf("pre_tip_glom.pdf")
plot_tree(phy_obj, "treeonly", title=paste("number of taxa before tip_glom() =", ntaxa(phy_obj))) + coord_polar(theta="y")
#dev.off()

ntaxa(tip_glom(phy_obj,hcfun = hclust))
#pdf("post_tip_glom.pdf")
plot_tree(tip_glom(phy_obj,hcfun = hclust),"treeonly", title=paste("number of taxa after tip_glom() =", ntaxa(phy_obj))) + coord_polar(theta="y") 
#dev.off()

phy_obj <- tip_glom(phy_obj,hcfun = hclust)
phy_obj
```

# FUNGuild analysis
The script was run as follows:
conda activate qiime2-2019.1
python <path/to/python/script> -otu <path/to/txt/file> -db fungi -m -u
conda deactivate

Reading FUNGuild output
``` {r}
guilds <- read.table("metagenomic/SeedPAD_OTU.guilds.txt", sep = "\t", header = TRUE, row.names = "OTU.ID")
guilds <- guilds %>% dplyr::select(taxonomy:Citation.Source)
guilds[,"ASV"] <- rownames(guilds)
head(guilds)
```
## Diversity analysis
Library size and normalization
```{r}
#Library size
sums <- sample_sums(phy_obj)
barplot(sums[order(sums, decreasing=TRUE)], las=3, cex.names = 0.5, ylab="Library size", col = "skyblue")
abline(h = 10000, lty=2)

#NOTE: sample P12211_1084 (S.Ctrl.4) has 11 reads. It will be removed from the analysis

#Normalisation by proportions
phy_obj <- prune_samples(sample_sums(phy_obj)>10000,phy_obj)
phy_obj

phy_norm <- transform_sample_counts(phy_obj,function(x) x/sum(x))
barplot(sample_sums(phy_norm),las=3,cex.names=.5,col="skyblue")

#Rarefy
set.seed(12345)
phy_raref <- rarefy_even_depth(phy_obj, sample.size = min(sample_sums(phy_obj)), replace = FALSE, trimOTUs = TRUE, verbose = TRUE)
barplot(sample_sums(phy_raref), las=3, cex.names = 0.5, col = "skyblue")
```

## Richness, alpha and beta diversity
### Richness 
Richness was calculated using the pd() function from picante package.
Preparing richness dataframe
```{r}
faiths_div <- pd(as.data.frame(t(otu_table(phy_raref))), phy_tree(phy_obj), include.root = FALSE)
faiths_div <- merge(faiths_div,sample_data(phy_obj), by = "row.names")
faiths_div$Fert_type <- factor(faiths_div$Fert_type, levels=c('Control','Arginine_P','Osmocote'))
faiths_div$Sample_type <- factor(faiths_div$Sample_type, levels=c('Root','Ectorhizosphere','Soil'))
```

Ploting richness
```{r echo= FALSE}
#pdf("Manuscript/Plots/Supplementary/richness_all.pdf")
ggplot(faiths_div, aes(x = Sample_type, y = SR, fill = Fert_type)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("green", "red","blue","indianred1","indianred2","indianred3","indianred4"), labels = c('Control','Arginine phosphate','Ammonium nitrate')) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Specie richness",limits = c(0,100))
```

Statistical analysis of richness using Kruskal-Wallis. Any significan differences were further analysed with Dunn’s test with dunnTest() function from FAS package.
The model used was richness~Sample type, richness~ fertilization type and richness~sample*fertilization
```{r}
aggregate(faiths_div$SR, list(faiths_div$Sample_type), median)
kruskal.test(SR ~ Sample_type, data = faiths_div) 
tmp <- dunnTest(SR ~ Sample_type, data = faiths_div, method = "bonferroni")
tmp <- tmp$res
cldList(comparison = tmp$Comparison, p.value = tmp$P.adj)

aggregate(faiths_div$SR, list(faiths_div$Fert_type), median)
kruskal.test(SR ~ Fert_type, data = faiths_div) #non significant
#tmp <- dunnTest(SR ~ Fert_type, data = faiths_div, method = "bonferroni")
#tmp <- tmp$res
#cldList(comparison = tmp$Comparison, p.value = tmp$P.adj)

aggregate(faiths_div$SR, list(faiths_div$fert_sample), median)
kruskal.test(SR ~ fert_sample, data = faiths_div) 
tmp <- dunnTest(SR ~ fert_sample, data = faiths_div, method = "bonferroni")
tmp <- tmp$res
cldList(comparison = tmp$Comparison, p.value = tmp$P.adj)
```

### Alpha diversity

The Alpha diversity was calculated using the Shannon diversity index with the function diversity() from the vegan package. 

Calculating Shannon diversity
```{r}
shannon_div <- diversity(otu_table(phy_obj), MARGIN = 2, index = "shannon")
shannon_div <- as.data.frame(shannon_div)
shannon_div <- merge(shannon_div,sample_data(phy_obj), by = "row.names")
shannon_div$Fert_type <- factor(shannon_div$Fert_type, levels=c('Control','Arginine_P','Osmocote','Edge_T','Edge_B','Edge_L','Edge_R'))
shannon_div$Sample_type <- factor(shannon_div$Sample_type, levels=c('Root','Ectorhizosphere','Soil','Edge'))
```

Plotting alpha diversity index
```{r echo= FALSE}
#pdf("Manuscript/Plots/Supplementary/Alpha_all.pdf")
ggplot(shannon_div, aes(x = Sample_type, y = shannon_div, fill = Fert_type)) +
  stat_boxplot(geom = "errorbar") +
  geom_boxplot(size = 1) + 
  scale_fill_manual(values=c("green", "red","blue","indianred1","indianred2","indianred3","indianred4"), labels = c('Control','Arginine phosphate','Ammonium nitrate','Top','Bottom','Left','Right')) +
  theme_bw() + 
  theme(text=element_text(size = 18),
        axis.text.x = element_text(size = 16, angle = 45, hjust = 1, colour = "black"),
        axis.text.y = element_text(size = 16, colour = "black"),
        axis.title.x = element_blank(),
        panel.border = element_rect(size = 2, fill = NA)) +
  scale_y_continuous(name = "Shannon Diversity Index",limits = c(0,4))
#dev.off()
```

Statistical analysis of Shannon diversity index using Kruskal-Wallis. Any significan differences were further analysed with Dunn’s test with dunnTest() function from FAS package.
The model used was shannon~Sample type, shannon~fertilization type and shannon~sample*fertilization
```{r}
aggregate(shannon_div$shannon_div, list(faiths_div$Sample_type), median)
kruskal.test(shannon_div ~ Sample_type, data = shannon_div)
tmp <- dunnTest(shannon_div ~ Sample_type, data = shannon_div, method = "bonferroni")
tmp <- tmp$res
cldList(comparison = tmp$Comparison, p.value = tmp$P.adj)

aggregate(shannon_div$shannon_div, list(faiths_div$Fert_type), median)
kruskal.test(shannon_div ~ Fert_type, data = shannon_div) #non-significant. no need of post-Hoc

aggregate(shannon_div$shannon_div, list(faiths_div$fert_sample), median)
kruskal.test(shannon_div ~ fert_sample, data = shannon_div) 
tmp <- dunnTest(shannon_div ~ fert_sample, data = shannon_div, method = "bonferroni")
tmp <- tmp$res
cldList(comparison = tmp$Comparison, p.value = tmp$P.adj)
```

```{r}
sessionInfo()
```